components/google_sign_in_permission/google_sign_in_permission_util.cc

```diff
@@ -62,7 +62,7 @@ bool IsGoogleAuthRelatedRequest(const GURL& request_url,
          IsGoogleAuthUrl(request_url) &&
          !IsGoogleAuthUrl(request_initiator_url) &&
          !net::registry_controlled_domains::SameDomainOrHost(
-             request_initiator_url, GURL(GetGoogleAuthPattern().GetHost()),
+             request_initiator_url, GURL(GetGoogleAuthPattern().ToString()),
```

Comment:
Let's just create a separate variable and check it here. It looks sketchy and bug-prone to convert a full pattern into a GURL.
```
bool IsGoogleAuthRelatedRequest(const GURL& request_url,
                                const GURL& request_initiator_url) {
  static const base::NoDestructor<GURL> kGoogleAuthPattern([] {
    DCHECK(!GetGoogleAuthPattern().HasDomainWildcard());
    DCHECK(!GetGoogleAuthPattern().GetHost().empty());
    return GURL("https://" + GetGoogleAuthPattern().GetHost());
  }());
  return ...
```

Reply:
I like that